CLOUDSTACK-8689: Verify effect of changing value of XenServer Max guest limit

pritisarap12 opened the pull request:
Verify effect of changing value of "Max guest limit" of XenServer on previously added hosts
    -Check if it reads changed value and act accordingly

sanju1010 commented on test/integration/testpaths/testpath_hypervisor_limit.py:
```diff
@@ -0,0 +1,191 @@
+                serviceofferingid=self.service_offering.id,
+                zoneid=self.zone.id,
+            )
+
```
If there are more than one host in a cluster, how can you make sure that all the vm deployments will go to the same host to test this functionality? IMO , it would be better to tag the host and try to deploy vms on it.

sanju1010 replied:
Also we need to check if there are any vms running on a chosen host(including system vms and guest vms) and then try to deploy the vms until we reach the max supported value.

pritisarap12 replied:
Added the suggested changes 
    -Verify all VM's go to same host
    -Count previously existing VM's on the selected host and then modify its "maxguestlimit"

nitt10prashant commented:
can you post test results ?

pritisarap12 commented:
Modified test to check for empty vmList and updating maxguestlimit accordingly.
And the test result is-

Test hypervisor max limits effect ... === TestName: test_check_hypervisor_max_limit_effect | Status : SUCCESS ===
ok

----------------------------------------------------------------------
Ran 1 test in 142.770s

OK

sanju1010 commented on test/integration/testpaths/testpath_hypervisor_limit.py:
```diff
@@ -0,0 +1,198 @@
+        vmCount = len(listVm)
+        # Increament vm count by 4(2 SSVMs 1 VR VM and 1 deployed VM)
+        newValue = vmCount + 4
+
```
If we assume that there are 2 SSVMs and 1 VR on the host then vmCount becomes 3. You are adding 4 to vmCount. So newValue becomes 7. That means max. guest limit on the host is 7.

pritisarap12 replied:
listVirtualMachines API does not list 2 SSVMs and 1 VR it only list the User deployed VM's hence I am adding 4 to vmCount.

sanju1010 commented on test/integration/testpaths/testpath_hypervisor_limit.py:
```diff
@@ -0,0 +1,198 @@
+            updateHypervisorCapabilitiesCmd()
+        cmdUpdate.id = self.hostCapId
+        cmdUpdate.maxguestslimit = newValue
+        self.apiclient.updateHypervisorCapabilities(cmdUpdate)
```
Here you are updating the Hypervisor Capabilities with new value with in our e.g. is 7. So in this case how you are expecting that the 2nd vm deployment would fail?

pritisarap12 commented:
Updated the testcase for:
     -Count SSVM's on selected host
     -Count VR on the host
     -Count user VM's on the host
     -Then update the maxguestlimit for that host as(ssvm+vr+user_vm+1)
     -Added validateList for host list

pritisarap12 commented:
Testcase Result-

Test hypervisor max limits effect ... === TestName: test_check_hypervisor_max_limit_effect | Status : SUCCESS ===
ok

Ran 1 test in 159.787s

OK

nitt10prashant commented:
LGTM
